### ISSUE TITLE
Original Purpose Minecarts

### ISSUE THREAD

**Issue by reyak** [StardewModders/mod-ideas#272]
So, minecarts are used as transport for you, but they're really supposed to transport stone and stuff. There are tons of minecarts in Stardew Valley, including unused ones with coal in the mines. My suggestion is simple: add an inventory gui to unload while you're in the mines, or wherever, so you can clean out your inventory, then grab it back when you need to from another location. Basically, the whole minecart system would have an extra option besides locations, which would open the new gui.

**Comment by KeepFireFlying** [comment]
So sort of like making the minecarts an inter-connected chest system?

**Comment by reyak** [comment]
Yes, a bit like the Minecraft Enderchest, or the Runescape coal carts.

**Comment by KeepFireFlying** [comment]
I'll have to look into those. I don't have much modding experience, so this might be beyond me, but I'll take a stab at it. In the meantime, anyone else who wants to grab this, feel free!

**Comment by Entoarox** [comment]
Intercept the "minecart activated" action (See Extended Minecarts for a means to do that) then trigger the menu for a chest that you hide in a inaccessible point on the Farm or FarmHouse map, blam, you have this mod. :)

Hope this helps.

**Comment by KeepFireFlying** [comment]
Thanks! I'll definitely take a look!
